std/meta.d

```diff
@@ -1309,6 +1309,52 @@ private template staticMerge(alias cmp, int half, Seq...)
     }
 }
 
+/**
+ * Allows subsetting of multiple members of tuple types that are not
+ * adjacent to each other, i.e. cannot be sliced.
+ *
+ * Evaluates to $(D AliasSeq!(T[range[0]], T[range[1]], ..., T[range[n]])).
+ */
+
+template staticSubset(alias range, T...)
+{
+	import std.range : isInputRange;
+	import std.traits : isArray, isNarrowString;
+
+	alias ArrT = typeof(range);
+	static if (isArray!ArrT && !isNarrowString!ArrT)
+	{
+		static if(range.length == 0)
+		{
+			alias staticSubset = AliasSeq!();
+		} else {
+			alias staticSubset = AliasSeq!(T[range[0]], staticSubset!(range[1 .. $], T));
```

Comment:
If I add template constraints as you describe were shall the import for isInputRange, and ElementType be placed. I guess they will have to be placed just before the template declaration:

```
import std.range : isInputRange;
import std.traits : isArray, isNarrowString, isUnsigned;
import std.range.primitives: ElementType;

template staticSubset(alias range, T...)
if(isInputRange!(typeof(range)) && is(ElementType!(typeof(range)) : size_t)) ...
```